force-app/main/default/classes/DMLOperations.cls

```diff
@@ -179,15 +253,38 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
+        Contact contactUpsert;
+
 		// Loop through each Contact
+        for (Contact contact : contactsToUpsert) {
+            contactUpsert = new Contact(LastName = contact.LastName, FirstName = contact.FirstName);
```

Comment:
contactsToUpsert is a List of existing Contact records, so we shouldn't need to create a new Contact when looping over them.